### ISSUE TITLE
Joy axis out of size, ipega controller on android

### ISSUE THREAD

**Issue by alglez** [godotengine/godot#11295]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Android, Ipega gamepad controller, Godot 2.1

**Issue description:**

Testing ipega gamepad joystick axis on android produces the following errors, just moving the joystick. It happens with left and right thumbstick, L2 and R2, and Dpad.

` ERROR: joy_axis: Index p_axis out of size (JOY_AXIS_MAX). `

One of the thumbstick (the left one) is recognized as `JOYSTICK_MOTION` event type, and it can print the axis `print(event.axis)`, but the other one is not recognized as such, even though error described above.

The input state of this gamepad is 

```~  adb shell dumpsys input state


ipega media gamepad controller
      Classes: 0x8000014b
      Path: /dev/input/event10
      Descriptor: 57d79ded5d0b57063f7a056ba57b4ea5681f295f
      Location: 
      ControllerNumber: 1
      UniqueId: 6C:65:B4:E1:90:00
      Identifier: bus=0x0005, vendor=0x1949, product=0x0402, version=0x011b
      KeyLayoutFile: /system/usr/keylayout/Generic.kl
      KeyCharacterMapFile: /system/usr/keychars/Generic.kcm
      ConfigurationFile: 
      HaveKeyboardLayoutOverlay: false


  Device 8: ipega media gamepad controller
    Generation: 2610
    IsExternal: true
    HasMic:     false
    Sources: 0x01002513
    KeyboardType: 2
    Motion Ranges:
      X: source=0x00002002, min=0.000, max=767.000, flat=0.000, fuzz=0.000, resolution=0.000
      Y: source=0x00002002, min=0.000, max=1023.000, flat=0.000, fuzz=0.000, resolution=0.000
      PRESSURE: source=0x00002002, min=0.000, max=1.000, flat=0.000, fuzz=0.000, resolution=0.000
      VSCROLL: source=0x00002002, min=-1.000, max=1.000, flat=0.000, fuzz=0.000, resolution=0.000
      X: source=0x01000010, min=-1.000, max=1.000, flat=0.118, fuzz=0.000, resolution=0.000
      Y: source=0x01000010, min=-1.000, max=1.000, flat=0.118, fuzz=0.000, resolution=0.000
      Z: source=0x01000010, min=-1.000, max=1.000, flat=0.118, fuzz=0.000, resolution=0.000
      RZ: source=0x01000010, min=-1.000, max=1.000, flat=0.118, fuzz=0.000, resolution=0.000
      GAS: source=0x01000010, min=0.000, max=1.000, flat=0.059, fuzz=0.000, resolution=0.000
      BRAKE: source=0x01000010, min=0.000, max=1.000, flat=0.059, fuzz=0.000, resolution=0.000
      HAT_X: source=0x01000010, min=-1.000, max=1.000, flat=0.000, fuzz=0.000, resolution=0.000
      HAT_Y: source=0x01000010, min=-1.000, max=1.000, flat=0.000, fuzz=0.000, resolution=0.000
    Keyboard Input Mapper:
      Parameters:
        HasAssociatedDisplay: false
        OrientationAware: false
        HandlesKeyRepeat: false
      KeyboardType: 2
      Orientation: 0
      KeyDowns: 0 keys currently down
      MetaState: 0x0
      DownTime: 1407787943983000
    Cursor Input Mapper:
      Parameters:
        HasAssociatedDisplay: true
        Mode: pointer
        OrientationAware: false
      XScale: 1.000
      YScale: 1.000
      XPrecision: 1.000
      YPrecision: 1.000
      HaveVWheel: true
      HaveHWheel: false
      VWheelScale: 1.000
      HWheelScale: 1.000
      Orientation: 0
      ButtonState: 0x00000000
      Down: false
      DownTime: 0
    Joystick Input Mapper:
      Axes:
        X: min=-1.00000, max=1.00000, flat=0.11765, fuzz=0.00000, resolution=0.00000
          scale=0.00784, offset=-1.00000, highScale=0.00784, highOffset=-1.00000
          rawAxis=0, rawMin=0, rawMax=255, rawFlat=15, rawFuzz=0, rawResolution=0
        Y: min=-1.00000, max=1.00000, flat=0.11765, fuzz=0.00000, resolution=0.00000
          scale=0.00784, offset=-1.00000, highScale=0.00784, highOffset=-1.00000
          rawAxis=1, rawMin=0, rawMax=255, rawFlat=15, rawFuzz=0, rawResolution=0
        Z: min=-1.00000, max=1.00000, flat=0.11765, fuzz=0.00000, resolution=0.00000
          scale=0.00784, offset=-1.00000, highScale=0.00784, highOffset=-1.00000
          rawAxis=2, rawMin=0, rawMax=255, rawFlat=15, rawFuzz=0, rawResolution=0
        RZ: min=-1.00000, max=1.00000, flat=0.11765, fuzz=0.00000, resolution=0.00000
          scale=0.00784, offset=-1.00000, highScale=0.00784, highOffset=-1.00000
          rawAxis=5, rawMin=0, rawMax=255, rawFlat=15, rawFuzz=0, rawResolution=0
        GAS: min=0.00000, max=1.00000, flat=0.05882, fuzz=0.00000, resolution=0.00000
          scale=0.00392, offset=0.00000, highScale=0.00392, highOffset=0.00000
          rawAxis=9, rawMin=0, rawMax=255, rawFlat=15, rawFuzz=0, rawResolution=0
        BRAKE: min=0.00000, max=1.00000, flat=0.05882, fuzz=0.00000, resolution=0.00000
          scale=0.00392, offset=0.00000, highScale=0.00392, highOffset=0.00000
          rawAxis=10, rawMin=0, rawMax=255, rawFlat=15, rawFuzz=0, rawResolution=0
        HAT_X: min=-1.00000, max=1.00000, flat=0.00000, fuzz=0.00000, resolution=0.00000
          scale=1.00000, offset=-0.00000, highScale=1.00000, highOffset=-0.00000
          rawAxis=16, rawMin=-1, rawMax=1, rawFlat=0, rawFuzz=0, rawResolution=0
        HAT_Y: min=-1.00000, max=1.00000, flat=0.00000, fuzz=0.00000, resolution=0.00000
          scale=1.00000, offset=-0.00000, highScale=1.00000, highOffset=-0.00000
          rawAxis=17, rawMin=-1, rawMax=1, rawFlat=0, rawFuzz=0, rawResolution=0
```

**Comment by akien-mga** [comment]
@Hinsbart I guess this may have been fixed by #12743? Otherwise please move it to the 3.1 milestone.

**Comment by alglez** [comment]
I don't have the controller with me right now, so I will test if it's fixed as soon as i can.

**Comment by reduz** [comment]
Any progress on this?

**Comment by akien-mga** [comment]
Assuming that it was the same issue as #24759, fixed by #24778.

Please comment if you can still reproduce the issue.
